4a.allosigma_filtering/allosigma-filtering

```
@@ -285,8 +285,90 @@ def filter_on_distance_value_biopython(PDB, df, distance_threshold, file):
+        raise ValueError("Only single-chain structures can be supported.")
+    #dictionary to store residue id and corresponding atom objects
+    residue_atoms = {}
+    for residue in structure.get_residues():
+        if not Bio.PDB.is_aa(residue):
+            continue
+        #list to store atom objects 
+        atoms = []
+        #handle glycine exception
+        if residue.get_resname() == "GLY":
+            H_atoms = [atom for atom in residue if atom.get_name() in {"HA2","HA3"}]
+            if len(H_atoms) != 2:
+                raise ValueError(
+                    f"Incomplete structure. Glycine at position "
+                    f"{residue.get_id()[1]} is missing H atoms."
+                    )
+            atoms.extend(H_atoms)
+        #handle rest of the residues
+        else:              
+            for atom in residue:
+                if atom.element != "H" and atom.get_name() not in {"N", "CA", "C", "O"}:
+                    atoms.append(atom)
+            if not atoms:
+                raise ValueError(
+                    f"Incomplete structure. Residue at position "
+                    f"{residue.get_id()[1]} is missing side-chain atoms."
+                    )
+        #store residue ID and atoms
+        residue_id = f"{Bio.PDB.Polypeptide.three_to_one(residue.get_resname())}{residue.get_id()[1]}"
+        residue_atoms[residue_id] = atoms
+
+    #add a column to the df with the position of the mutation
+    df['position'] = df.apply(lambda r: r.name.split(' ')[0][:-1], axis=1)
+    #use this as a index
+    df = df.reset_index()
+    df.set_index('position', inplace=True)
+    #iterates over the df checking for non-NaN entries  
+    for res1 in df.index:
+        for res2 in df.columns:
+            if pd.isna(df.loc[res1, res2]):
+                continue
+            atoms1 = residue_atoms.get(res1)
+            atoms2 = residue_atoms.get(res2)
+            if atoms1 and atoms2:
```

this check is not useful; there should be no cases in which we have 0 atoms (because of check in 335). And either way, we'd rather the program to crash rather than just skip silently if that was the case

---

True, I’ve removed the if atoms1 and atoms2 check